modules/server/src/network/index.ts

```diff
@@ -35,16 +35,16 @@ const fetchRemoteSchema = async (
 			gqlRequest: { query: getIntrospectionQuery() },
 		});
 
-		if (response.status !== 200) {
-			throw Error('network error');
-		}
-
-		const jsonData = await response.data.json();
-
-		if (jsonData && jsonData.data && isGqlIntrospectionQuery(jsonData.data)) {
-			return { config, introspectionResult: jsonData };
+		if (response.status === 200 && response.statusText === 'OK') {
+			// axios response "data" field, graphql response "data" field
+			const responseData = response.data?.data;
+			if (isGqlIntrospectionQuery(responseData)) {
+				return { config, introspectionResult: responseData };
+			} else {
+				throw Error('response data unexpected');
+			}
 		} else {
-			throw Error('response data unexpected');
+			throw Error('network error');
```

Comment:
when I throw at any point in the logic, I tend to also log.debug out what function is the one that's throwing
...hopefully in another PR later on we can add winston/equivalent as discussed over Slack